pkg/controller/nodenetworkconfigurationpolicy/conditions/enactments.go

```diff
@@ -0,0 +1,87 @@
+package conditions
+
+import (
+	corev1 "k8s.io/api/core/v1"
+
+	nmstatev1alpha1 "github.com/nmstate/kubernetes-nmstate/pkg/apis/nmstate/v1alpha1"
+)
+
+func setEnactmentFailedToConfigure(enactments *nmstatev1alpha1.EnactmentList, nodeName string, message string) {
+	setEnactmentFailed(enactments, nodeName, nmstatev1alpha1.NodeNetworkConfigurationEnactmentConditionFailedToConfigure, message)
```

Comment:
maybe names shouldn't have version as part of them ?

unless version isnt going to change

Reply:
This is a must in golang, so you are able to have part of the code using different api versions.